File: client/src/app/models/todo.ts
import { Priority } from "../enums/priority";

export class Todo {
    #id: number;
    #title: string;
    #description: string;
    #publishDate : Date;
    #deadLineDate : Date;
    #priority : Priority;
    #isDone: boolean;

    constructor(id:number, title: string, description: string, publishDate : Date, deadLineDate : Date, priority : Priority, isDone: boolean){
        this.#id = id,
        this.#title=title,
        this.#description=description,
        this.#publishDate = publishDate,
        this.#deadLineDate = deadLineDate,
        this.#priority = priority,
        this.#isDone = isDone
    }

    switchDateTypeToString(date: Date): string {
        const formatter = new Intl.DateTimeFormat('fr', { dateStyle: 'short' });
        return formatter.format(date);
    }

    changeIsDoneState(): void{
        this.#isDone = !this.#isDone;
    }

     get id(): number {
        return this.#id;
    }

    get title(): string {
        return this.#title;
    }

    get description(): string {
        return this.#description;
    }

    get publishDate(): Date {
        return this.#publishDate;
    }

    get deadLineDate(): Date {
        return this.#deadLineDate;
    }

    get priority(): Priority {
        return this.#priority;
    }

    get isDone(): boolean {
        return this.#isDone;
    }

    set title(value: string) {
        this.#title = value;
    }

    set description(value: string) {
        this.#description = value;
    }

    set publishDate(value: Date) {
        this.#publishDate = value;
    }

    set deadLineDate(value: Date) {
        this.#deadLineDate = value;
    }

    set priority(value: Priority) {
        this.#priority = value;
    }

    set isDone(value: boolean) {
        this.#isDone = value;
    }

 


}

File: server/src/main/java/com/productivityapp/demo/domain/entity/Todo.java
package com.productivityapp.demo.domain.entity;

import com.productivityapp.demo.domain.enumeration.Priority;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
@Entity
public class Todo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @NotBlank
    @Size(min = 1,max = 50,message = "title has to contain 1 to 50 characters")
    private String title;

    @Size(max = 500,message = "the description has to contain 500 characters maximum")
    private String description;

    @NotNull
    private LocalDate publishDate;

    @NotNull(message = "dead line date must be filled this way : dd/mm/yyyy")
    private LocalDate deadLineDate;

    @NotNull(message = "there must be a priority")
    private Priority priority;

    @NotNull
    private boolean isDone;

}
